src/panoramic/cli/connections.py

```diff
@@ -0,0 +1,317 @@
+
+    new_connection = connections[name].copy()
+    _update_connection_from_args(
+        new_connection,
+        type=type,
+        user=user,
+        password=password,
+        host=host,
+        port=port,
+        database=database,
+        schema=schema,
+        warehouse=warehouse,
+        account=account,
+        project=project,
+        key_file=key_file,
+    )
+
+    credentials, error = get_dialect_credentials(new_connection)
+    if error is not None:
+        raise ConnectionUpdateException(error)
+
+    if not no_test:
+        ok, error = Connections.test(credentials)
+        if not ok:
+            raise ConnectionUpdateException(error)
+
+    # Empty values are important for DBT credentials verification, but we don't need to store empty values in config.
+    for key, value in new_connection.items():
+        if value != '':
+            connections[name][key] = value
+
+    Connections.save(connections)
+    echo_info('Connection was successfully updated!')
+
+
+def list_connections_command(show_password: bool) -> None:
+    """CLI command. List all connections."""
+    connections = Connections.load()
+    if not connections:
+        config_file = Paths.config_file()
+        echo_info(
+            f'No connections found.\n'
+            f'Use "pano connection create" to create connection or edit "{config_file}" file.'
+        )
+        exit(0)
+
+    if not show_password:
+        for conn in connections.values():
+            conn['password'] = '*****'
+    echo_info(yaml.dump(connections))
+
+
+def remove_connection_command(name: str) -> None:
+    """CLI command. Remove existing connection."""
+    connections = Connections.load()
+    if name not in connections:
+        raise ConnectionNotFound(name)
+
+    del connections[name]
+    Connections.save(connections)
+
+
+def test_connections_command(name: Optional[str] = None) -> None:
+    """CLI command. Test connections by trying to connect to the database.
+    Optionally you can specify name for specific connection to that only that."""
+    connections = Connections.load()
+    if name is not None:
+        if name not in connections:
+            raise ConnectionNotFound(name)
+        # Filter specified connection by name
+        connections = {name: connections[name]}
+
+    for name, connection in connections.items():
+        # Fill with default values because DBT requires some fields we don't.
+        for key in CONNECTION_KEYS:
+            if key not in connection:
+                connection[key] = ''
+
+        credentials, error = get_dialect_credentials(connection)
+        if error is not None:
+            echo_info(f'{name}... FAIL: {error}')
+            continue
+
+        ok, error = Connections.test(credentials)
+        if ok:
+            echo_info(f'{name}... OK')
+        else:
+            echo_info(f'{name}... FAIL: {error}')
+
+
+class Connections:
+    def __init__(self):
+        self._connections = self.load()
```

Comment:
Are the `__init__` and `get_by_name_cached` used anywhere? Can we get rid of them if not and we want to use the static methods on `Connections`?

Reply:
Done